python/TestHarness/validation/ValidationCase.py

```diff
@@ -164,6 +164,8 @@ def __post_init__(self):
                 assert len(self.bounds) == 2
                 assert isinstance(self.bounds[0], type(self.value))
                 assert isinstance(self.bounds[1], type(self.value))
+            if self.rel_err is not None:
+                assert isinstance(self.rel_err, float)
```

Comment:
Would it make sense to throw errors if:
- the validation flag is provided, but neither the bounds nor the relative error is added. Or to make sure that they are not provided at the same time.

Reply:
https://github.com/idaholab/moose/blob/49310948a8ce01f5eb8c1a1c87ad6968b88fcf31/python/TestHarness/validation/ValidationCase.py#L574

This ensures that a test case needs results